src/optics/focusAtom.ts

```diff
@@ -0,0 +1,74 @@
+import { atom } from 'jotai'
+import * as O from 'optics-ts'
+import type { PrimitiveAtom, SetStateAction, WritableAtom } from '../core/types'
+
+export type FocusAtom = {
+  <S, A>(
+    atom: PrimitiveAtom<S>,
+    callback: (optic: O.OpticFor<S>) => O.Prism<S, any, A>
+  ): WritableAtom<A | undefined, SetStateAction<A>>
```

Comment:
When does it become `undefined` ? It's preferable if this is avoidable as A can include undefined.

Reply:
So it will return a `WritableAtom<A | undefined, SetStateAction<A>>` whenever a `Prism` is returned.

A prism can for example be: When getting an index out of an array `['hello', 'there'][5]` you never know if the type is `string` or `undefined`.

Another example of using prism would be when using `guard`, https://github.com/akheron/optics-ts#guardu-extends-ag-a-a--a-is-u-prisms-_-u

Reply:
Oh, I see. So, a prism is like an array, `A` only represents the array item type. Got it.